server/konnectors/captain_train.js

```diff
@@ -228,7 +228,7 @@ function fetchBills(requiredFields, entries, data, next) {
     let linkedPNR = [data.pnrs.find(pnr => pnr.id === proof.pnr_id)];
     try {
       linkedPNR = data.pnrs.filter(
-        pnr => pnr.proof_ids.indexOf(proof.id) !== -1
+        pnr => typeof pnr.proof_ids !== 'undefined' && pnr.proof_ids.indexOf(proof.id) !== -1
```

Comment:
Do you expect an array ? Maybe `instanceof Array` would be easier to read than !== 'undefined' ?